assignments/tests/test_views.py

```diff
@@ -0,0 +1,86 @@
+import pytest
+from django.shortcuts import reverse
+from rest_framework.test import APIClient
+
+from assignments.models import Assignment, BudgetingTarget, BudgetingTask, OpenTextTask, Section
+
+
+class TestApi:
+    @pytest.mark.django_db
+    def test_assignment_endpoint_returns_only_open_assignments(self, create_assignments, assignments_url):
+        assert Assignment.objects.count() == 2
+        api_client = APIClient()
+        response = api_client.get(assignments_url)
+        response_data = response.json()
+        assert len(response_data) == 1
+        assert response_data[0]['status'] == Assignment.STATUS_OPEN
+
+    @pytest.mark.django_db
+    def test_all_sections_present_in_open_assignment(self, create_assignments, assignments_url):
+        assignment = Assignment.objects.get(status=Assignment.STATUS_OPEN)
+        api_client = APIClient()
+        response = api_client.get(reverse('assignments:section-list', args=[assignment.slug]))
+        response_data = response.json()
+        assert len(response_data) == assignment.sections.count()
+        for section in response_data:
+            assert section['title'] in assignment.sections.values_list('title', flat=True)
+
+    @pytest.mark.django_db
+    def test_all_open_text_tasks_present_in_section_for_open_assignment(self, create_assignments, assignments_url):
+        assignment = Assignment.objects.get(status=Assignment.STATUS_OPEN)
+        api_client = APIClient()
+        open_text_tasks = OpenTextTask.objects.filter(section__assignment=assignment)
+        for task in open_text_tasks:
+            section_url = reverse('assignments:section-detail', args=[assignment.slug, task.section.id])
+            response = api_client.get(section_url)
+            response_data = response.json()
+            found = False
+            for data_open_text_task in response_data['opentexttasks']:
```

Comment:
I think I could set different name in Serializer data, and set source to be opentexttasks. Is that better?